src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static animalsFilter() {
+    Animal.alive = Animal.alive.filter(animal => animal.health > 0
+    || animal.health === 'undefined');
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        Animal.animalsFilter();
```

Comment:
Calling 'Animal.animalsFilter()' inside a condition is not optimal. It would be more efficient to call it after the 'bite' operation, as it needs to be done regardless of whether the animal's health reaches 0 or not.